rust/main/agents/relayer/src/msg/metadata/base.rs

```diff
@@ -187,3 +192,189 @@ impl AppContextClassifier {
         Ok(None)
     }
 }
+
+/// An ISM caching policy.
+#[derive(Copy, Clone, Debug, PartialEq, Default, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum IsmCachePolicy {
+    /// Default cache policy, includes the message in the cache key
+    /// when querying config that may be message-specific.
+    #[default]
+    MessageSpecific,
```

Comment:
worth clarifying in a comment that `MessageSpecific` is the default because it makes the fewest assumptions about the mutability of the ISM result